<a href="https://colab.research.google.com/github/nnarey/My-Projects/blob/main/FIFA_World_Rankings_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Ranking Valuation:
Top 3 Tournaments Will be chosen for teams

###Tier 1 Tournaments: (x4 Multiplier)

- FIFA World Cup

###Tier 2 Tournaments: (x3 Multiplier)

- UEFA Euro
- CONMEBOL Copa America

###Tier 3 Tournaments: (x2 Multiplier)

- UEFA Nations League
- CAF Cup of Nations
- AFC Asian Cup

###Tier 4 Tournaments: (No Multiplier)

- CONCACAF Gold Cup
- CONCACAF Nations League
- World Cup Qualifiers (Final Stage)

###Tier 5 Tournaments (0.5x Miltiplier)
- AFC World Cup Qualifying (Second Round to be replaced with third round when it happens)

# CONCACAF

### 2023 Gold Cup

In [120]:
# CONCACAF GOLD CUP 2023
import requests
from bs4 import BeautifulSoup
import pandas as pd

url = "https://fbref.com/en/comps/681/stats/Gold-Cup-Stats"

def scrape_football_reference(url):
    response = requests.get(url)
    if response.status_code == 200:
        soup = BeautifulSoup(response.text, 'html.parser')
        table = soup.find('table')
        df = pd.read_html(str(table))[0]
        return df
    else: # Print an error message if the request was not successful
        print('Error: Unable to retrieve data from the provided URL')
        return None

df_CCF = scrape_football_reference(url)
df_CCF

<ipython-input-120-2586d79aea43>:13: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  df = pd.read_html(str(table))[0]


Unnamed: 0_level_0 Unnamed: 1_level_0 Unnamed: 2_level_0  \
                   Squad               # Pl                Age   
0              ca Canada                 22               27.5   
1          cr Costa Rica                 19               27.2   
2                cu Cuba                 20               25.6   
3         sv El Salvador                 19               26.2   
4          gp Guadeloupe                 17               28.4   
5           gt Guatemala                 18               27.4   
6               ht Haiti                 18               26.8   
7            hn Honduras                 20               27.9   
8             jm Jamaica                 22               27.1   
9          mq Martinique                 19               27.4   
10             mx Mexico                 20               27.0   
11             pa Panama                 22               28.5   
12              qa Qatar                 22               25.0   
13  kn St. Kitts & Nevis                 21               27.8   
14      tt Trin & Tobago                 22               28.2   
15      us United States                 22               24.5   

   Unnamed: 3_level_0 Playing Time                  Performance      ...       \
                 Poss           MP Starts  Min  90s         Gls Ast  ... G-PK   
0                51.8            4     44  390  4.3           7   3  ...    5   
1                49.0            4     44  360  4.0           6   4  ...    5   
2                47.7            3     33  270  3.0           3   0  ...    0   
3                51.3            3     33  270  3.0           3   2  ...    2   
4                45.7            3     33  270  3.0           7   5  ...    6   
5                51.5            4     44  360  4.0           4   3  ...    4   
6                43.0            3     33  270  3.0           4   2  ...    3   
7                55.3            3     33  270  3.0           3   3  ...    3   
8                48.2            5     55  450  5.0          10   8  ...   10   
9                36.3            3     33  270  3.0           7   3  ...    6   
10               58.8            6     66  540  6.0          12   7  ...   11   
11               59.3            6     66  570  6.3          11  10  ...   11   
12               37.0            4     44  360  4.0           3   3  ...    3   
13               34.0            3     33  270  3.0           0   0  ...    0   
14               48.7            3     33  270  3.0           3   3  ...    3   
15               62.4            5     55  510  5.7          15  11  ...   14   

                      Per 90 Minutes                           
   PK PKatt CrdY CrdR            Gls   Ast   G+A  G-PK G+A-PK  
0   2     2   12    0           1.62  0.69  2.31  1.15   1.85  
1   1     1   10    0           1.50  1.00  2.50  1.25   2.25  
2   3     3    9    1           1.00  0.00  1.00  0.00   0.00  
3   1     2    3    0           1.00  0.67  1.67  0.67   1.33  
4   1     2    8    1           2.33  1.67  4.00  2.00   3.67  
5   0     1    8    0           1.00  0.75  1.75  1.00   1.75  
6   1     1    5    0           1.33  0.67  2.00  1.00   1.67  
7   0     0    5    1           1.00  1.00  2.00  1.00   2.00  
8   0     1   12    0           2.00  1.60  3.60  2.00   3.60  
9   1     1    6    1           2.33  1.00  3.33  2.00   3.00  
10  1     1   12    0           2.00  1.17  3.17  1.83   3.00  
11  0     0   11    0           1.74  1.58  3.32  1.74   3.32  
12  0     0   15    1           0.75  0.75  1.50  0.75   1.50  
13  0     0    7    0           0.00  0.00  0.00  0.00   0.00  
14  0     0    4    0           1.00  1.00  2.00  1.00   2.00  
15  1     1    4    0           2.65  1.94  4.59  2.47   4.41  

[16 rows x 21 columns]

In [121]:
df_CCF.columns = range(df_CCF.shape[1])
df_CCF = df_CCF.T.reset_index().T.reset_index(drop=True).rename(columns=lambda x: f'Column_{x}')
df_CCF = df_CCF.iloc[1:]

In [122]:
df_CCF = df_CCF.drop(columns=['Column_1', 'Column_5', 'Column_6', 'Column_7'])

In [123]:
df_CCF

Column_0 Column_2 Column_3 Column_4 Column_8 Column_9  \
1              ca Canada     27.5     51.8        4        7        3   
2          cr Costa Rica     27.2     49.0        4        6        4   
3                cu Cuba     25.6     47.7        3        3        0   
4         sv El Salvador     26.2     51.3        3        3        2   
5          gp Guadeloupe     28.4     45.7        3        7        5   
6           gt Guatemala     27.4     51.5        4        4        3   
7               ht Haiti     26.8     43.0        3        4        2   
8            hn Honduras     27.9     55.3        3        3        3   
9             jm Jamaica     27.1     48.2        5       10        8   
10         mq Martinique     27.4     36.3        3        7        3   
11             mx Mexico     27.0     58.8        6       12        7   
12             pa Panama     28.5     59.3        6       11       10   
13              qa Qatar     25.0     37.0        4        3        3   
14  kn St. Kitts & Nevis     27.8     34.0        3        0        0   
15      tt Trin & Tobago     28.2     48.7        3        3        3   
16      us United States     24.5     62.4        5       15       11   

   Column_10 Column_11 Column_12 Column_13 Column_14 Column_15 Column_16  \
1         10         5         2         2        12         0      1.62   
2         10         5         1         1        10         0       1.5   
3          3         0         3         3         9         1       1.0   
4          5         2         1         2         3         0       1.0   
5         12         6         1         2         8         1      2.33   
6          7         4         0         1         8         0       1.0   
7          6         3         1         1         5         0      1.33   
8          6         3         0         0         5         1       1.0   
9         18        10         0         1        12         0       2.0   
10        10         6         1         1         6         1      2.33   
11        19        11         1         1        12         0       2.0   
12        21        11         0         0        11         0      1.74   
13         6         3         0         0        15         1      0.75   
14         0         0         0         0         7         0       0.0   
15         6         3         0         0         4         0       1.0   
16        26        14         1         1         4         0      2.65   

   Column_17 Column_18 Column_19 Column_20  
1       0.69      2.31      1.15      1.85  
2        1.0       2.5      1.25      2.25  
3        0.0       1.0       0.0       0.0  
4       0.67      1.67      0.67      1.33  
5       1.67       4.0       2.0      3.67  
6       0.75      1.75       1.0      1.75  
7       0.67       2.0       1.0      1.67  
8        1.0       2.0       1.0       2.0  
9        1.6       3.6       2.0       3.6  
10       1.0      3.33       2.0       3.0  
11      1.17      3.17      1.83       3.0  
12      1.58      3.32      1.74      3.32  
13      0.75       1.5      0.75       1.5  
14       0.0       0.0       0.0       0.0  
15       1.0       2.0       1.0       2.0  
16      1.94      4.59      2.47      4.41

In [124]:
df_CCF = df_CCF.rename(columns={'Column_0': 'Nation', 'Column_2': 'Average Age', 'Column_3': 'Possession %', 'Column_4': 'Matches Played',
                                'Column_8': 'Goals', 'Column_9': 'Assists', 'Column_10': 'G+A', 'Column_11': 'G-Pk', 'Column_12': 'Pk', 'Column_13': 'Pk Att',
                                'Column_14':'Yellow Cards', 'Column_15':'Red Cards', 'Column_16': 'G/90',
                                'Column_17': 'A/90', 'Column_18': 'G+A/90','Column_19':'G-Pk/90', 'Column_20':'G+A-Pk/90'})
df_CCF

Nation Average Age Possession % Matches Played Goals  \
1              ca Canada        27.5         51.8              4     7   
2          cr Costa Rica        27.2         49.0              4     6   
3                cu Cuba        25.6         47.7              3     3   
4         sv El Salvador        26.2         51.3              3     3   
5          gp Guadeloupe        28.4         45.7              3     7   
6           gt Guatemala        27.4         51.5              4     4   
7               ht Haiti        26.8         43.0              3     4   
8            hn Honduras        27.9         55.3              3     3   
9             jm Jamaica        27.1         48.2              5    10   
10         mq Martinique        27.4         36.3              3     7   
11             mx Mexico        27.0         58.8              6    12   
12             pa Panama        28.5         59.3              6    11   
13              qa Qatar        25.0         37.0              4     3   
14  kn St. Kitts & Nevis        27.8         34.0              3     0   
15      tt Trin & Tobago        28.2         48.7              3     3   
16      us United States        24.5         62.4              5    15   

   Assists G+A G-Pk Pk Pk Att Yellow Cards Red Cards  G/90  A/90 G+A/90  \
1        3  10    5  2      2           12         0  1.62  0.69   2.31   
2        4  10    5  1      1           10         0   1.5   1.0    2.5   
3        0   3    0  3      3            9         1   1.0   0.0    1.0   
4        2   5    2  1      2            3         0   1.0  0.67   1.67   
5        5  12    6  1      2            8         1  2.33  1.67    4.0   
6        3   7    4  0      1            8         0   1.0  0.75   1.75   
7        2   6    3  1      1            5         0  1.33  0.67    2.0   
8        3   6    3  0      0            5         1   1.0   1.0    2.0   
9        8  18   10  0      1           12         0   2.0   1.6    3.6   
10       3  10    6  1      1            6         1  2.33   1.0   3.33   
11       7  19   11  1      1           12         0   2.0  1.17   3.17   
12      10  21   11  0      0           11         0  1.74  1.58   3.32   
13       3   6    3  0      0           15         1  0.75  0.75    1.5   
14       0   0    0  0      0            7         0   0.0   0.0    0.0   
15       3   6    3  0      0            4         0   1.0   1.0    2.0   
16      11  26   14  1      1            4         0  2.65  1.94   4.59   

   G-Pk/90 G+A-Pk/90  
1     1.15      1.85  
2     1.25      2.25  
3      0.0       0.0  
4     0.67      1.33  
5      2.0      3.67  
6      1.0      1.75  
7      1.0      1.67  
8      1.0       2.0  
9      2.0       3.6  
10     2.0       3.0  
11    1.83       3.0  
12    1.74      3.32  
13    0.75       1.5  
14     0.0       0.0  
15     1.0       2.0  
16    2.47      4.41

In [125]:
df_CCF['Nation'] = df_CCF['Nation'].str[2:]
df_CCF

Nation Average Age Possession % Matches Played Goals Assists  \
1               Canada        27.5         51.8              4     7       3   
2           Costa Rica        27.2         49.0              4     6       4   
3                 Cuba        25.6         47.7              3     3       0   
4          El Salvador        26.2         51.3              3     3       2   
5           Guadeloupe        28.4         45.7              3     7       5   
6            Guatemala        27.4         51.5              4     4       3   
7                Haiti        26.8         43.0              3     4       2   
8             Honduras        27.9         55.3              3     3       3   
9              Jamaica        27.1         48.2              5    10       8   
10          Martinique        27.4         36.3              3     7       3   
11              Mexico        27.0         58.8              6    12       7   
12              Panama        28.5         59.3              6    11      10   
13               Qatar        25.0         37.0              4     3       3   
14   St. Kitts & Nevis        27.8         34.0              3     0       0   
15       Trin & Tobago        28.2         48.7              3     3       3   
16       United States        24.5         62.4              5    15      11   

   G+A G-Pk Pk Pk Att Yellow Cards Red Cards  G/90  A/90 G+A/90 G-Pk/90  \
1   10    5  2      2           12         0  1.62  0.69   2.31    1.15   
2   10    5  1      1           10         0   1.5   1.0    2.5    1.25   
3    3    0  3      3            9         1   1.0   0.0    1.0     0.0   
4    5    2  1      2            3         0   1.0  0.67   1.67    0.67   
5   12    6  1      2            8         1  2.33  1.67    4.0     2.0   
6    7    4  0      1            8         0   1.0  0.75   1.75     1.0   
7    6    3  1      1            5         0  1.33  0.67    2.0     1.0   
8    6    3  0      0            5         1   1.0   1.0    2.0     1.0   
9   18   10  0      1           12         0   2.0   1.6    3.6     2.0   
10  10    6  1      1            6         1  2.33   1.0   3.33     2.0   
11  19   11  1      1           12         0   2.0  1.17   3.17    1.83   
12  21   11  0      0           11         0  1.74  1.58   3.32    1.74   
13   6    3  0      0           15         1  0.75  0.75    1.5    0.75   
14   0    0  0      0            7         0   0.0   0.0    0.0     0.0   
15   6    3  0      0            4         0   1.0   1.0    2.0     1.0   
16  26   14  1      1            4         0  2.65  1.94   4.59    2.47   

   G+A-Pk/90  
1       1.85  
2       2.25  
3        0.0  
4       1.33  
5       3.67  
6       1.75  
7       1.67  
8        2.0  
9        3.6  
10       3.0  
11       3.0  
12      3.32  
13       1.5  
14       0.0  
15       2.0  
16      4.41

In [126]:
# Add in Rows For Points

df_CCF["GC Points"] = [6, 4, 0, 2, 4, 7, 3, 4, 10, 3, 15, 11, 4, 0, 3, 9]
df_CCF = df_CCF.sort_values(by='GC Points', ascending=False)
df_CCF

Nation Average Age Possession % Matches Played Goals Assists  \
11              Mexico        27.0         58.8              6    12       7   
12              Panama        28.5         59.3              6    11      10   
9              Jamaica        27.1         48.2              5    10       8   
16       United States        24.5         62.4              5    15      11   
6            Guatemala        27.4         51.5              4     4       3   
1               Canada        27.5         51.8              4     7       3   
2           Costa Rica        27.2         49.0              4     6       4   
5           Guadeloupe        28.4         45.7              3     7       5   
8             Honduras        27.9         55.3              3     3       3   
13               Qatar        25.0         37.0              4     3       3   
7                Haiti        26.8         43.0              3     4       2   
10          Martinique        27.4         36.3              3     7       3   
15       Trin & Tobago        28.2         48.7              3     3       3   
4          El Salvador        26.2         51.3              3     3       2   
3                 Cuba        25.6         47.7              3     3       0   
14   St. Kitts & Nevis        27.8         34.0              3     0       0   

   G+A G-Pk Pk Pk Att Yellow Cards Red Cards  G/90  A/90 G+A/90 G-Pk/90  \
11  19   11  1      1           12         0   2.0  1.17   3.17    1.83   
12  21   11  0      0           11         0  1.74  1.58   3.32    1.74   
9   18   10  0      1           12         0   2.0   1.6    3.6     2.0   
16  26   14  1      1            4         0  2.65  1.94   4.59    2.47   
6    7    4  0      1            8         0   1.0  0.75   1.75     1.0   
1   10    5  2      2           12         0  1.62  0.69   2.31    1.15   
2   10    5  1      1           10         0   1.5   1.0    2.5    1.25   
5   12    6  1      2            8         1  2.33  1.67    4.0     2.0   
8    6    3  0      0            5         1   1.0   1.0    2.0     1.0   
13   6    3  0      0           15         1  0.75  0.75    1.5    0.75   
7    6    3  1      1            5         0  1.33  0.67    2.0     1.0   
10  10    6  1      1            6         1  2.33   1.0   3.33     2.0   
15   6    3  0      0            4         0   1.0   1.0    2.0     1.0   
4    5    2  1      2            3         0   1.0  0.67   1.67    0.67   
3    3    0  3      3            9         1   1.0   0.0    1.0     0.0   
14   0    0  0      0            7         0   0.0   0.0    0.0     0.0   

   G+A-Pk/90  GC Points  
11       3.0         15  
12      3.32         11  
9        3.6         10  
16      4.41          9  
6       1.75          7  
1       1.85          6  
2       2.25          4  
5       3.67          4  
8        2.0          4  
13       1.5          4  
7       1.67          3  
10       3.0          3  
15       2.0          3  
4       1.33          2  
3        0.0          0  
14       0.0          0

### 2024 Nations League

In [127]:
df_CCF["NL Points"] = [9, 6, 6, 15, 0, 3, 0, 0, 3, 0, 0, 0, 3, 0, 0, 0]
df_CCF["COPA Points"] = [4, 6, 0, 3, 0, 5, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0]
df_CCF["WC Points"] = [4, 0, 0, 5, 0, 0, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0]
df_CCF["WC Points"] = df_CCF["WC Points"] * 4
df_CCF["T2 Points"] = df_CCF["COPA Points"] * 3
df_CCF["Total Points"] = 0.5 * (df_CCF["GC Points"] + df_CCF["NL Points"]) + df_CCF["COPA Points"] + df_CCF["WC Points"]
df_CCF = df_CCF.sort_values(by='Total Points', ascending=False)
df_CCF["T3 Points"] = (df_CCF["GC Points"] + df_CCF["NL Points"]) * 0.5
df_CCF["Total Points"] = df_CCF["WC Points"] + df_CCF["T2 Points"] + df_CCF["T3 Points"]
df_CCF = df_CCF[["Nation", "WC Points", "T2 Points", "T3 Points", "Total Points"]]
df_CCF

Nation  WC Points  T2 Points  T3 Points  Total Points
16       United States         20          9       12.0          41.0
11              Mexico         16         12       12.0          40.0
2           Costa Rica         12         12        2.0          26.0
12              Panama          0         18        8.5          26.5
1               Canada          0         15        4.5          19.5
9              Jamaica          0          0        8.0           8.0
6            Guatemala          0          0        3.5           3.5
8             Honduras          0          0        3.5           3.5
15       Trin & Tobago          0          0        3.0           3.0
5           Guadeloupe          0          0        2.0           2.0
13               Qatar          0          0        2.0           2.0
7                Haiti          0          0        1.5           1.5
10          Martinique          0          0        1.5           1.5
4          El Salvador          0          0        1.0           1.0
3                 Cuba          0          0        0.0           0.0
14   St. Kitts & Nevis          0          0        0.0           0.0

# UEFA

In [128]:
url = "https://fbref.com/en/comps/676/stats/UEFA-Euro-Stats"
df_UEFA = scrape_football_reference(url)
df_UEFA

<ipython-input-120-2586d79aea43>:13: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  df = pd.read_html(str(table))[0]


Unnamed: 0_level_0 Unnamed: 1_level_0 Unnamed: 2_level_0  \
                Squad               # Pl                Age   
0          al Albania                 19               27.0   
1          at Austria                 19               27.0   
2          be Belgium                 18               27.8   
3          hr Croatia                 19               28.6   
4          cz Czechia                 21               26.8   
5          dk Denmark                 19               28.1   
6         eng England                 21               26.6   
7           fr France                 18               26.7   
8          ge Georgia                 20               26.7   
9          de Germany                 23               29.0   
10         hu Hungary                 20               27.3   
11           it Italy                 21               26.7   
12     nl Netherlands                 21               27.0   
13          pl Poland                 22               27.5   
14        pt Portugal                 24               28.1   
15         ro Romania                 21               27.8   
16       sct Scotland                 18               28.2   
17          rs Serbia                 19               27.5   
18        sk Slovakia                 20               29.8   
19        si Slovenia                 19               26.9   
20           es Spain                 25               26.9   
21     ch Switzerland                 20               29.3   
22         tr Türkiye                 23               26.1   
23         ua Ukraine                 23               25.5   

   Unnamed: 3_level_0 Playing Time                  Performance      ...  \
                 Poss           MP Starts  Min  90s         Gls Ast  ...   
0                35.7            3     33  270  3.0           3   1  ...   
1                52.8            4     44  360  4.0           6   5  ...   
2                55.0            4     44  360  4.0           2   1  ...   
3                55.7            3     33  270  3.0           2   1  ...   
4                40.7            3     33  270  3.0           3   2  ...   
5                54.0            4     44  360  4.0           2   2  ...   
6                54.9            7     77  690  7.7           8   5  ...   
7                50.8            6     66  570  6.3           2   1  ...   
8                33.8            4     44  360  4.0           4   2  ...   
9                62.6            5     55  480  5.3          11   8  ...   
10               41.3            3     33  270  3.0           2   2  ...   
11               53.8            4     44  360  4.0           3   2  ...   
12               53.7            6     66  540  6.0           9   7  ...   
13               41.3            3     33  270  3.0           3   1  ...   
14               66.6            5     55  510  5.7           3   1  ...   
15               38.0            4     44  360  4.0           4   2  ...   
16               44.3            3     33  270  3.0           1   1  ...   
17               51.3            3     33  270  3.0           1   1  ...   
18               47.8            4     44  390  4.3           4   3  ...   
19               32.3            4     44  390  4.3           2   1  ...   
20               58.1            7     77  660  7.3          14  13  ...   
21               47.0            5     55  480  5.3           8   5  ...   
22               49.4            5     55  450  5.0           8   6  ...   
23               51.3            3     33  270  3.0           2   2  ...   

   Per 90 Minutes                                                             
              Gls   Ast   G+A  G-PK G+A-PK    xG   xAG xG+xAG  npxG npxG+xAG  
0            1.00  0.33  1.33  1.00   1.33  0.81  0.53   1.34  0.81     1.34  
1            1.50  1.25  2.75  1.25   2.50  1.73  1.31   3.04  1.53     2.85  
2            0.50  0.25  0.75  0.50   0.75  1.12  0.78   1.89  1.12    

In [129]:
df_UEFA.columns = range(df_UEFA.shape[1])
df_UEFA = df_UEFA.T.reset_index().T.reset_index(drop=True).rename(columns=lambda x: f'Column_{x}')
df_UEFA = df_UEFA.iloc[1:]
df_UEFA

Column_0 Column_1 Column_2 Column_3 Column_4 Column_5 Column_6  \
1       al Albania       19     27.0     35.7        3       33      270   
2       at Austria       19     27.0     52.8        4       44      360   
3       be Belgium       18     27.8     55.0        4       44      360   
4       hr Croatia       19     28.6     55.7        3       33      270   
5       cz Czechia       21     26.8     40.7        3       33      270   
6       dk Denmark       19     28.1     54.0        4       44      360   
7      eng England       21     26.6     54.9        7       77      690   
8        fr France       18     26.7     50.8        6       66      570   
9       ge Georgia       20     26.7     33.8        4       44      360   
10      de Germany       23     29.0     62.6        5       55      480   
11      hu Hungary       20     27.3     41.3        3       33      270   
12        it Italy       21     26.7     53.8        4       44      360   
13  nl Netherlands       21     27.0     53.7        6       66      540   
14       pl Poland       22     27.5     41.3        3       33      270   
15     pt Portugal       24     28.1     66.6        5       55      510   
16      ro Romania       21     27.8     38.0        4       44      360   
17    sct Scotland       18     28.2     44.3        3       33      270   
18       rs Serbia       19     27.5     51.3        3       33      270   
19     sk Slovakia       20     29.8     47.8        4       44      390   
20     si Slovenia       19     26.9     32.3        4       44      390   
21        es Spain       25     26.9     58.1        7       77      660   
22  ch Switzerland       20     29.3     47.0        5       55      480   
23      tr Türkiye       23     26.1     49.4        5       55      450   
24      ua Ukraine       23     25.5     51.3        3       33      270   

   Column_7 Column_8 Column_9  ... Column_22 Column_23 Column_24 Column_25  \
1       3.0        3        1  ...       1.0      0.33      1.33       1.0   
2       4.0        6        5  ...       1.5      1.25      2.75      1.25   
3       4.0        2        1  ...       0.5      0.25      0.75       0.5   
4       3.0        2        1  ...      0.67      0.33       1.0      0.67   
5       3.0        3        2  ...       1.0      0.67      1.67       1.0   
6       4.0        2        2  ...       0.5       0.5       1.0       0.5   
7       7.7        8        5  ...      1.04      0.65       1.7      0.91   
8       6.3        2        1  ...      0.32      0.16      0.47      0.16   
9       4.0        4        2  ...       1.0       0.5       1.5       0.5   
10      5.3       11        8  ...      2.06       1.5      3.56      1.69   
11      3.0        2        2  ...      0.67      0.67      1.33      0.67   
12      4.0        3        2  ...      0.75       0.5      1.25      0.75   
13      6.0        9        7  ...       1.5      1.17      2.67       1.5   
14      3.0        3        1  ...       1.0      0.33      1.33      0.67   
15      5.7        3        1  ...      0.53      0.18      0.71      0.53   
16      4.0        4        2  ...       1.0       0.5       1.5      0.75   
17      3.0        1        1  ...      0.33      0.33      0.67      0.33   
18      3.0        1        1  ...      0.33      0.33      0.67      0.33   
19      4.3        4        3  ...      0.92      0.69      1.62      0.92   
20      4.3        2        1  ...      0.46      0.23      0.69      0.46   
21      7.3       14       13  ...      1.91      1.77      3.68      1.91   
22      5.3        8        5  ...       1.5      0.94      2.44       1.5   
23      5.0        8        6  ...       1.6       1.2       2.8       1.6   
24      3.0        2        2  ...      0.67      0.67      1.33      0.67   

   Column_26 Column_27 Column_28 Column_29 Column_30 Column_31  
1       1.33      0.81      0.53      1.34      0.81      1.34  
2        2.5      1.73      

In [130]:
df_UEFA = df_UEFA.drop(columns=['Column_1', 'Column_5', 'Column_6', 'Column_7'])
df_UEFA = df_UEFA.rename(columns={'Column_0': 'Nation', 'Column_2': 'Average Age', 'Column_3': 'Possession %', 'Column_4': 'Matches Played',
                                'Column_8': 'Goals', 'Column_9': 'Assists', 'Column_10': 'G+A', 'Column_11': 'G-Pk', 'Column_12': 'Pk', 'Column_13': 'Pk Att',
                                'Column_14':'Yellow Cards', 'Column_15':'Red Cards', 'Column_16': 'G/90',
                                'Column_17': 'A/90', 'Column_18': 'G+A/90','Column_19':'G-Pk/90', 'Column_20':'G+A-Pk/90'})
df_UEFA

Nation Average Age Possession % Matches Played Goals Assists G+A  \
1       al Albania        27.0         35.7              3     3       1   4   
2       at Austria        27.0         52.8              4     6       5  11   
3       be Belgium        27.8         55.0              4     2       1   3   
4       hr Croatia        28.6         55.7              3     2       1   3   
5       cz Czechia        26.8         40.7              3     3       2   5   
6       dk Denmark        28.1         54.0              4     2       2   4   
7      eng England        26.6         54.9              7     8       5  13   
8        fr France        26.7         50.8              6     2       1   3   
9       ge Georgia        26.7         33.8              4     4       2   6   
10      de Germany        29.0         62.6              5    11       8  19   
11      hu Hungary        27.3         41.3              3     2       2   4   
12        it Italy        26.7         53.8              4     3       2   5   
13  nl Netherlands        27.0         53.7              6     9       7  16   
14       pl Poland        27.5         41.3              3     3       1   4   
15     pt Portugal        28.1         66.6              5     3       1   4   
16      ro Romania        27.8         38.0              4     4       2   6   
17    sct Scotland        28.2         44.3              3     1       1   2   
18       rs Serbia        27.5         51.3              3     1       1   2   
19     sk Slovakia        29.8         47.8              4     4       3   7   
20     si Slovenia        26.9         32.3              4     2       1   3   
21        es Spain        26.9         58.1              7    14      13  27   
22  ch Switzerland        29.3         47.0              5     8       5  13   
23      tr Türkiye        26.1         49.4              5     8       6  14   
24      ua Ukraine        25.5         51.3              3     2       2   4   

   G-Pk Pk Pk Att  ... Column_22 Column_23 Column_24 Column_25 Column_26  \
1     3  0      0  ...       1.0      0.33      1.33       1.0      1.33   
2     5  1      1  ...       1.5      1.25      2.75      1.25       2.5   
3     2  0      0  ...       0.5      0.25      0.75       0.5      0.75   
4     2  0      2  ...      0.67      0.33       1.0      0.67       1.0   
5     3  0      0  ...       1.0      0.67      1.67       1.0      1.67   
6     2  0      0  ...       0.5       0.5       1.0       0.5       1.0   
7     7  1      1  ...      1.04      0.65       1.7      0.91      1.57   
8     1  1      1  ...      0.32      0.16      0.47      0.16      0.32   
9     2  2      2  ...       1.0       0.5       1.5       0.5       1.0   
10    9  2      2  ...      2.06       1.5      3.56      1.69      3.19   
11    2  0      0  ...      0.67      0.67      1.33      0.67      1.33   
12    3  0      0  ...      0.75       0.5      1.25      0.75      1.25   
13    9  0      0  ...       1.5      1.17      2.67       1.5      2.67   
14    2  1      1  ...       1.0      0.33      1.33      0.67       1.0   
15    3  0      1  ...      0.53      0.18      0.71      0.53      0.71   
16    3  1      1  ...       1.0       0.5       1.5      0.75      1.25   
17    1  0      0  ...      0.33      0.33      0.67      0.33      0.67   
18    1  0      0  ...      0.33      0.33      0.67      0.33      0.67   
19    4  0      0  ...      0.92      0.69      1.62      0.92      1.62   
20    2  0      0  ...      0.46      0.23      0.69      0.46      0.69   
21   14  0      0  ...      1.91      1.77      3.68      1.91      3.68   
22    8  0      0  ...       1.5      0.94      2.44       1.5      2.44   
23    8  0      0  ...       1.6       1.2       2.8       1.6       2.8   
24    2  0      0  ...      0.67      0.67      1.33      0.67      1.33   

   Column_27 Column_28 Column_29 Column_30 Column_31  
1       0.81      0.53      1.34      0.81      1.34  


In [131]:
df_UEFA['Nation'] = df_UEFA['Nation'].str[3:]
df_UEFA['T2 Points'] = [1, 6, 4, 2, 1, 3, 10, 9, 4, 11, 3, 4, 10, 1, 8, 4, 1, 2, 5, 4, 19, 9, 9, 4]
df_UEFA['T2 Points'] = df_UEFA['T2 Points'] * 3
df_UEFA["WC Points"] = [0, 0, 4, 7, 0, 1, 10, 16, 0, 4, 0, 0, 11, 4, 9, 0, 0, 1, 0, 0, 5, 6, 0, 0]
df_UEFA['WC Points'] = df_UEFA['WC Points'] * 4
df_UEFA["T3 Points"] = [0, 4, 10, 17, 4, 12, 3, 5, 0, 7, 10, 11, 16, 7, 10, 0, 0, 0, 0, 0, 15, 9, 0, 0]
df_UEFA["T3 Points"] = df_UEFA["T3 Points"] * 2
df_UEFA['Total Points'] = (df_UEFA['T2 Points']) + (df_UEFA['WC Points']) + df_UEFA['T3 Points']
df_UEFA = df_UEFA.sort_values(by='Total Points', ascending=False)
df_UEFA = df_UEFA[['Nation', 'WC Points', 'T2 Points', 'T3 Points', 'Total Points']]
df_UEFA

Nation  WC Points  T2 Points  T3 Points  Total Points
21        Spain         20         57         30           107
13  Netherlands         44         30         32           106
8        France         64         27         10           101
15     Portugal         36         24         20            80
7       England         40         30          6            76
22  Switzerland         24         27         18            69
4       Croatia         28          6         34            68
10      Germany         16         33         14            63
3       Belgium         16         12         20            48
6       Denmark          4          9         24            37
12        Italy          0         12         22            34
14       Poland         16          3         14            33
11      Hungary          0          9         20            29
23      Türkiye          0         27          0            27
2       Austria          0         18          8            26
19     Slovakia          0         15          0            15
20     Slovenia          0         12          0            12
24      Ukraine          0         12          0            12
16      Romania          0         12          0            12
9       Georgia          0         12          0            12
5       Czechia          0          3          8            11
18       Serbia          4          6          0            10
17     Scotland          0          3          0             3
1       Albania          0          3          0             3

In [132]:
merged_df = pd.concat([df_CCF, df_UEFA], ignore_index=True)
merged_df = merged_df.sort_values(by='Total Points', ascending=False)
merged_df['Rank'] = merged_df['Total Points'].rank(method='dense', ascending=False)
merged_df["Rank"] = merged_df["Rank"].astype(int)
merged_df = merged_df.set_index('Rank')
merged_df

Nation  WC Points  T2 Points  T3 Points  Total Points
Rank                                                                   
1                  Spain         20         57       30.0         107.0
2            Netherlands         44         30       32.0         106.0
3                 France         64         27       10.0         101.0
4               Portugal         36         24       20.0          80.0
5                England         40         30        6.0          76.0
6            Switzerland         24         27       18.0          69.0
7                Croatia         28          6       34.0          68.0
8                Germany         16         33       14.0          63.0
9                Belgium         16         12       20.0          48.0
10         United States         20          9       12.0          41.0
11                Mexico         16         12       12.0          40.0
12               Denmark          4          9       24.0          37.0
13                 Italy          0         12       22.0          34.0
14                Poland         16          3       14.0          33.0
15               Hungary          0          9       20.0          29.0
16               Türkiye          0         27        0.0          27.0
17                Panama          0         18        8.5          26.5
18               Austria          0         18        8.0          26.0
18            Costa Rica         12         12        2.0          26.0
19                Canada          0         15        4.5          19.5
20              Slovakia          0         15        0.0          15.0
21               Georgia          0         12        0.0          12.0
21               Romania          0         12        0.0          12.0
21               Ukraine          0         12        0.0          12.0
21              Slovenia          0         12        0.0          12.0
22               Czechia          0          3        8.0          11.0
23                Serbia          4          6        0.0          10.0
24               Jamaica          0          0        8.0           8.0
25             Guatemala          0          0        3.5           3.5
25              Honduras          0          0        3.5           3.5
26              Scotland          0          3        0.0           3.0
26               Albania          0          3        0.0           3.0
26         Trin & Tobago          0          0        3.0           3.0
27            Guadeloupe          0          0        2.0           2.0
27                 Qatar          0          0        2.0           2.0
28                 Haiti          0          0        1.5           1.5
28            Martinique          0          0        1.5           1.5
29           El Salvador          0          0        1.0           1.0
30     St. Kitts & Nevis          0          0        0.0           0.0
30                  Cuba          0          0        0.0           0.0

### Current Rankings
1. Netherlands 106
2. France 101
3. Spain 98
4. Portugal 80
5. England 76
6. Switzerland 69
8. Croatia 68
7. Germany 63
10. Belgium 48
9. USA 39.5
10. Mexico 38
14. Denmark 37
17. Italy 34
13. Poland 33
19. Hungary 29
12. Turkiye 27
18. Austria 26
15. Costa Rica 24
16. Panama 23.5
20. Slovakia 15

# CONMEBOL

In [133]:
data = {
    'Nation': ['Argentina', 'Bolivia', 'Brazil', 'Chile', 'Colombia', 'Ecuador', 'Paraguay', 'Peru', 'Uruguay', 'Venezuela'],
    'WC Points': [14, 0, 10, 0, 0, 4, 0, 0, 4, 0],
    'COPA Points': [14, 0, 6, 2, 14, 5, 0, 1, 10, 10],
    'WCQ Points': [15, 3, 7, 5, 12, 8, 5, 2, 13, 9],
}

df_CONM = pd.DataFrame(data)
df_CONM

Nation  WC Points  COPA Points  WCQ Points
0  Argentina         14           14          15
1    Bolivia          0            0           3
2     Brazil         10            6           7
3      Chile          0            2           5
4   Colombia          0           14          12
5    Ecuador          4            5           8
6   Paraguay          0            0           5
7       Peru          0            1           2
8    Uruguay          4           10          13
9  Venezuela          0           10           9

In [134]:
df_CONM['Total Points'] = (df_CONM['WC Points'] * 4) + (df_CONM['COPA Points'] * 3) + df_CONM['WCQ Points']
df_CONM['WC Points'] = df_CONM['WC Points'] * 4
df_CONM['T2 Points'] = df_CONM['COPA Points'] * 3
df_CONM['T3 Points'] = df_CONM['WCQ Points']
df_CONM = df_CONM[['Nation', 'WC Points', 'T2 Points', 'T3 Points', 'Total Points']]
df_CONM = df_CONM.sort_values(by='Total Points', ascending=False)
df_CONM

Nation  WC Points  T2 Points  T3 Points  Total Points
0  Argentina         56         42         15           113
2     Brazil         40         18          7            65
8    Uruguay         16         30         13            59
4   Colombia          0         42         12            54
5    Ecuador         16         15          8            39
9  Venezuela          0         30          9            39
3      Chile          0          6          5            11
6   Paraguay          0          0          5             5
7       Peru          0          3          2             5
1    Bolivia          0          0          3             3

In [135]:
merged_df = pd.concat([merged_df, df_CONM], ignore_index=True)
merged_df = merged_df.sort_values(by='Total Points', ascending=False)
merged_df['Rank'] = merged_df['Total Points'].rank(method='dense', ascending=False)
merged_df["Rank"] = merged_df["Rank"].astype(int)
merged_df = merged_df.set_index('Rank')
merged_df

Nation  WC Points  T2 Points  T3 Points  Total Points
Rank                                                                   
1              Argentina         56         42       15.0         113.0
2                  Spain         20         57       30.0         107.0
3            Netherlands         44         30       32.0         106.0
4                 France         64         27       10.0         101.0
5               Portugal         36         24       20.0          80.0
6                England         40         30        6.0          76.0
7            Switzerland         24         27       18.0          69.0
8                Croatia         28          6       34.0          68.0
9                 Brazil         40         18        7.0          65.0
10               Germany         16         33       14.0          63.0
11               Uruguay         16         30       13.0          59.0
12              Colombia          0         42       12.0          54.0
13               Belgium         16         12       20.0          48.0
14         United States         20          9       12.0          41.0
15                Mexico         16         12       12.0          40.0
16               Ecuador         16         15        8.0          39.0
16             Venezuela          0         30        9.0          39.0
17               Denmark          4          9       24.0          37.0
18                 Italy          0         12       22.0          34.0
19                Poland         16          3       14.0          33.0
20               Hungary          0          9       20.0          29.0
21               Türkiye          0         27        0.0          27.0
22                Panama          0         18        8.5          26.5
23            Costa Rica         12         12        2.0          26.0
23               Austria          0         18        8.0          26.0
24                Canada          0         15        4.5          19.5
25              Slovakia          0         15        0.0          15.0
26               Ukraine          0         12        0.0          12.0
26              Slovenia          0         12        0.0          12.0
26               Romania          0         12        0.0          12.0
26               Georgia          0         12        0.0          12.0
27                 Chile          0          6        5.0          11.0
27               Czechia          0          3        8.0          11.0
28                Serbia          4          6        0.0          10.0
29               Jamaica          0          0        8.0           8.0
30                  Peru          0          3        2.0           5.0
30              Paraguay          0          0        5.0           5.0
31             Guatemala          0          0        3.5           3.5
31              Honduras          0          0        3.5           3.5
32               Albania          0          3        0.0           3.0
32              Scotland          0          3        0.0           3.0
32               Bolivia          0          0        3.0           3.0
32         Trin & Tobago          0          0        3.0           3.0
33                 Qatar          0          0        2.0           2.0
33            Guadeloupe          0          0        2.0           2.0
34            Martinique          0          0        1.5           1.5
34                 Haiti          0          0        1.5           1.5
35           El Salvador          0          0        1.0           1.0
36                  Cuba          0          0        0.0           0.0
36     St. Kitts & Nevis          0          0        0.0           0.0

### Updated
1. Agentina 113
3. Netherlands 106
2. France 101
4. Spain 98
6. Portugal 80
5. England 76
6. Switzerland 69
11. Croatia 68
7. Brazil 65
9. Germany 63
8. Uruguay 59
10. Colombia 54
12. USA 41
13. Mexico 40
14. Ecuador 39
14. Venezuela 39
16. Belgium 38
17. Turkiye 27
18. Panama 26.5
19. Poland 26
19. Costa Rica 26
21. Denmark 25
22. Italy 23
23. Austria 22
24. Hungary 19
25. Slovakia 15

# CAF

In [136]:
data = {
    'Nation': (['Angola', 'Namibia', 'Nigeria', 'Cameroon', 'Cape Verde',
                "Mauritania", 'Morocco', 'South Africa', 'Equitorial Guinea',
                'Guinea', 'Egypt', 'DR Congo', 'Senegal', 'Ivory Coast', 'Mali',
                'Burkina Faso', 'Ghana', 'Tunisia', 'Sudan', 'Rwanda', 'Algeria',
                'Comoros']),
    'WC Points': [0, 0, 0, 4, 0, 0, 11, 0, 0, 0, 0, 0, 6, 0, 0, 0, 3, 4, 0, 0, 0, 0],
    'T2 Points': [10, 4, 14, 4, 11, 3, 7, 9, 7, 7, 4, 7, 10, 13, 8, 4, 2, 2, 0, 0, 2, 0],
    'T3 Points': [6, 8, 3, 8, 7, 1, 12, 7, 3, 6, 10, 7, 8, 10, 5, 5, 9, 10, 10, 7, 9, 9],
}

df_CAF = pd.DataFrame(data)
df_CAF

Nation  WC Points  T2 Points  T3 Points
0              Angola          0         10          6
1             Namibia          0          4          8
2             Nigeria          0         14          3
3            Cameroon          4          4          8
4          Cape Verde          0         11          7
5          Mauritania          0          3          1
6             Morocco         11          7         12
7        South Africa          0          9          7
8   Equitorial Guinea          0          7          3
9              Guinea          0          7          6
10              Egypt          0          4         10
11           DR Congo          0          7          7
12            Senegal          6         10          8
13        Ivory Coast          0         13         10
14               Mali          0          8          5
15       Burkina Faso          0          4          5
16              Ghana          3          2          9
17            Tunisia          4          2         10
18              Sudan          0          0         10
19             Rwanda          0          0          7
20            Algeria          0          2          9
21            Comoros          0          0          9

In [137]:
df_CAF['WC Points'] = df_CAF['WC Points'] * 4
df_CAF['T2 Points'] = df_CAF['T2 Points'] * 2
df_CAF["Total Points"] = df_CAF["WC Points"] + df_CAF["T2 Points"] + df_CAF["T3 Points"]
df_CAF = df_CAF.sort_values(by='Total Points', ascending=False)
df_CAF

Nation  WC Points  T2 Points  T3 Points  Total Points
6             Morocco         44         14         12            70
12            Senegal         24         20          8            52
13        Ivory Coast          0         26         10            36
3            Cameroon         16          8          8            32
2             Nigeria          0         28          3            31
17            Tunisia         16          4         10            30
4          Cape Verde          0         22          7            29
0              Angola          0         20          6            26
7        South Africa          0         18          7            25
16              Ghana         12          4          9            25
14               Mali          0         16          5            21
11           DR Congo          0         14          7            21
9              Guinea          0         14          6            20
10              Egypt          0          8         10            18
8   Equitorial Guinea          0         14          3            17
1             Namibia          0          8          8            16
15       Burkina Faso          0          8          5            13
20            Algeria          0          4          9            13
18              Sudan          0          0         10            10
21            Comoros          0          0          9             9
5          Mauritania          0          6          1             7
19             Rwanda          0          0          7             7

In [138]:
merged_df = pd.concat([merged_df, df_CAF], ignore_index=True)
merged_df = merged_df.sort_values(by='Total Points', ascending=False)
merged_df['Rank'] = merged_df['Total Points'].rank(method='dense', ascending=False)
merged_df["Rank"] = merged_df["Rank"].astype(int)
merged_df = merged_df.set_index('Rank')
merged_df.head(40)

Nation  WC Points  T2 Points  T3 Points  Total Points
Rank                                                               
1          Argentina         56         42       15.0         113.0
2              Spain         20         57       30.0         107.0
3        Netherlands         44         30       32.0         106.0
4             France         64         27       10.0         101.0
5           Portugal         36         24       20.0          80.0
6            England         40         30        6.0          76.0
7            Morocco         44         14       12.0          70.0
8        Switzerland         24         27       18.0          69.0
9            Croatia         28          6       34.0          68.0
10            Brazil         40         18        7.0          65.0
11           Germany         16         33       14.0          63.0
12           Uruguay         16         30       13.0          59.0
13          Colombia          0         42       12.0          54.0
14           Senegal         24         20        8.0          52.0
15           Belgium         16         12       20.0          48.0
16     United States         20          9       12.0          41.0
17            Mexico         16         12       12.0          40.0
18           Ecuador         16         15        8.0          39.0
18         Venezuela          0         30        9.0          39.0
19           Denmark          4          9       24.0          37.0
20       Ivory Coast          0         26       10.0          36.0
21             Italy          0         12       22.0          34.0
22            Poland         16          3       14.0          33.0
23          Cameroon         16          8        8.0          32.0
24           Nigeria          0         28        3.0          31.0
25           Tunisia         16          4       10.0          30.0
26        Cape Verde          0         22        7.0          29.0
26           Hungary          0          9       20.0          29.0
27           Türkiye          0         27        0.0          27.0
28            Panama          0         18        8.5          26.5
29        Costa Rica         12         12        2.0          26.0
29           Austria          0         18        8.0          26.0
29            Angola          0         20        6.0          26.0
30      South Africa          0         18        7.0          25.0
30             Ghana         12          4        9.0          25.0
31              Mali          0         16        5.0          21.0
31          DR Congo          0         14        7.0          21.0
32            Guinea          0         14        6.0          20.0
33            Canada          0         15        4.5          19.5
34             Egypt          0          8       10.0          18.0

#AFC

In [139]:
data = {
    'Nation': (['Qatar', 'Tajikstan', 'Australia', 'Uzbekistan', 'Iran', 'UAE', 'Iraq', 'Japan', 'Bahrain', 'South Korea', 'Saudi Arabia', 'Thailand', 'Indonesia', 'Jordan', 'Palestine', 'Syria']),
    'WC Points': [0, 0, 6, 0, 3, 0, 0, 7, 0, 4, 3, 0, 0, 0, 0, 0],
    'T2 Points': [19, 5, 10, 9, 13, 5, 9, 9, 6, 9, 8, 5, 3, 13, 4, 5],
    'T3 Points': [8, 4, 9, 7, 7, 8, 9, 9, 5.5, 8, 6.5, 4, 5, 6.5, 4, 3.5],
}

df_AFC = pd.DataFrame(data)
df_AFC

Nation  WC Points  T2 Points  T3 Points
0          Qatar          0         19        8.0
1      Tajikstan          0          5        4.0
2      Australia          6         10        9.0
3     Uzbekistan          0          9        7.0
4           Iran          3         13        7.0
5            UAE          0          5        8.0
6           Iraq          0          9        9.0
7          Japan          7          9        9.0
8        Bahrain          0          6        5.5
9    South Korea          4          9        8.0
10  Saudi Arabia          3          8        6.5
11      Thailand          0          5        4.0
12     Indonesia          0          3        5.0
13        Jordan          0         13        6.5
14     Palestine          0          4        4.0
15         Syria          0          5        3.5

In [140]:
df_AFC['WC Points'] = df_AFC['WC Points'] * 4
df_AFC['T2 Points'] = df_AFC['T2 Points'] * 2
df_AFC["Total Points"] = df_AFC["WC Points"] + df_AFC["T2 Points"] + df_AFC["T3 Points"]
df_AFC = df_AFC.sort_values(by='Total Points', ascending=False)
df_AFC

Nation  WC Points  T2 Points  T3 Points  Total Points
7          Japan         28         18        9.0          55.0
2      Australia         24         20        9.0          53.0
0          Qatar          0         38        8.0          46.0
4           Iran         12         26        7.0          45.0
9    South Korea         16         18        8.0          42.0
10  Saudi Arabia         12         16        6.5          34.5
13        Jordan          0         26        6.5          32.5
6           Iraq          0         18        9.0          27.0
3     Uzbekistan          0         18        7.0          25.0
5            UAE          0         10        8.0          18.0
8        Bahrain          0         12        5.5          17.5
1      Tajikstan          0         10        4.0          14.0
11      Thailand          0         10        4.0          14.0
15         Syria          0         10        3.5          13.5
14     Palestine          0          8        4.0          12.0
12     Indonesia          0          6        5.0          11.0

In [141]:
merged_df = pd.concat([merged_df, df_AFC], ignore_index=True)
merged_df = merged_df.sort_values(by='Total Points', ascending=False)
merged_df['Rank'] = merged_df['Total Points'].rank(method='min', ascending=False)
merged_df["Rank"] = merged_df["Rank"].astype(int)
merged_df = merged_df.set_index('Rank')
merged_df = merged_df.head(30)
merged_df

Nation  WC Points  T2 Points  T3 Points  Total Points
Rank                                                               
1          Argentina         56         42       15.0         113.0
2              Spain         20         57       30.0         107.0
3        Netherlands         44         30       32.0         106.0
4             France         64         27       10.0         101.0
5           Portugal         36         24       20.0          80.0
6            England         40         30        6.0          76.0
7            Morocco         44         14       12.0          70.0
8        Switzerland         24         27       18.0          69.0
9            Croatia         28          6       34.0          68.0
10            Brazil         40         18        7.0          65.0
11           Germany         16         33       14.0          63.0
12           Uruguay         16         30       13.0          59.0
13             Japan         28         18        9.0          55.0
14          Colombia          0         42       12.0          54.0
15         Australia         24         20        9.0          53.0
16           Senegal         24         20        8.0          52.0
17           Belgium         16         12       20.0          48.0
18             Qatar          0         38        8.0          46.0
19              Iran         12         26        7.0          45.0
20       South Korea         16         18        8.0          42.0
21     United States         20          9       12.0          41.0
22            Mexico         16         12       12.0          40.0
23         Venezuela          0         30        9.0          39.0
23           Ecuador         16         15        8.0          39.0
25           Denmark          4          9       24.0          37.0
26       Ivory Coast          0         26       10.0          36.0
27      Saudi Arabia         12         16        6.5          34.5
28             Italy          0         12       22.0          34.0
29            Poland         16          3       14.0          33.0
30            Jordan          0         26        6.5          32.5

In [142]:
merged_df["FIFA Ranking"] = [1, 3, 7, 2, 8, 4, 14, 15, 12, 5, 13, 11, 18, 9, 24, 19, 6, 34, 20, 23, 16, 17, 37, 27, 21, 38, 56, 10, 28, 68]
merged_df

Nation  WC Points  T2 Points  T3 Points  Total Points  \
Rank                                                                  
1          Argentina         56         42       15.0         113.0   
2              Spain         20         57       30.0         107.0   
3        Netherlands         44         30       32.0         106.0   
4             France         64         27       10.0         101.0   
5           Portugal         36         24       20.0          80.0   
6            England         40         30        6.0          76.0   
7            Morocco         44         14       12.0          70.0   
8        Switzerland         24         27       18.0          69.0   
9            Croatia         28          6       34.0          68.0   
10            Brazil         40         18        7.0          65.0   
11           Germany         16         33       14.0          63.0   
12           Uruguay         16         30       13.0          59.0   
13             Japan         28         18        9.0          55.0   
14          Colombia          0         42       12.0          54.0   
15         Australia         24         20        9.0          53.0   
16           Senegal         24         20        8.0          52.0   
17           Belgium         16         12       20.0          48.0   
18             Qatar          0         38        8.0          46.0   
19              Iran         12         26        7.0          45.0   
20       South Korea         16         18        8.0          42.0   
21     United States         20          9       12.0          41.0   
22            Mexico         16         12       12.0          40.0   
23         Venezuela          0         30        9.0          39.0   
23           Ecuador         16         15        8.0          39.0   
25           Denmark          4          9       24.0          37.0   
26       Ivory Coast          0         26       10.0          36.0   
27      Saudi Arabia         12         16        6.5          34.5   
28             Italy          0         12       22.0          34.0   
29            Poland         16          3       14.0          33.0   
30            Jordan          0         26        6.5          32.5   

      FIFA Ranking  
Rank                
1                1  
2                3  
3                7  
4                2  
5                8  
6                4  
7               14  
8               15  
9               12  
10               5  
11              13  
12              11  
13              18  
14               9  
15              24  
16              19  
17               6  
18              34  
19              20  
20              23  
21              16  
22              17  
23              37  
23              27  
25              21  
26              38  
27              56  
28              10  
29              28  
30              68

# Analysis

The results from the ranking have some similarities to the FIFA World Rankings, but in large part differ as well. Here is a nation by nation analysis of the top 10 plus a few of choice:

1. Argentina: Both systems agree that the world champion is the rightful and far away best team on Earth currently.

W vs #4 France 3 (4)-(2) 3;
W vs #14 Colombia 1-0;
W vs #9 Croatia 3-0;
W vs #3 Netherlands 2 (4)-(3) 2;

2. Spain: The Spanish are looking like the next most powerful team in the world. Coming off disappointment in the World Cup, Spain has now won back to back European championships with the Euro and Nations League. Few would argue after they won every game in the Euro without relying on penalties that they are not the second best; FIFA currently ranks France, who lost to Spain in the semifinals of the Euro, over Spain.

W vs #6 England 2-1;
W vs #4 France 2-1;
W vs #9 Croatia 0 (5)-(4) 0;
L vs #7 Morocco 0 (0)-(3) 0;

3. Netherlands: The Dutch took Argentina to penalty kicks in the '22 World Cup and had semifinal appearances in both the Euro and the Nations League. Constantly in the almost champions column and consistently among the world's best. Their inability to reach glory is why FIFA ranks them down at 7th, but their consistency is why they rank so high with mine.

L vs #1 Argentina 2 (3)-(4) 2;
L vs #6 England 1-2;
L vs #9 Croatia 2-4;
W vs #16 Senegal 2-0;

4. France: The World Cup runner ups felt like they underperformed in the Euro and in the Nations League. France relied on extra time and penalty kicks to evade elimination, yet their evading elimination is why FIFA ranks them so high, and my system places them just below.

L vs #1 Argentina 3 (2)-(4) 3;
L vs #2 Spain 1-2;
W vs #7 Morocco 2-0;
W vs #6 England 2-1;

5. Portugal: This ranking may feel a little controversial. Portugal got eliminated in the quarter finals of the World Cup, fell one point short of Spain in the nations league to move on from the group, and lost once again in penalties to France in the quarter finals of the Euro. However, not being embarassed and having strong performances in each tournament still places them among the globe's best. Their inability to win in the clutch keeps them from moving up. FIFA ranks them 8th, below a Belgium that has underperformed at every tournament since the 2018 World Cup.

L vs #7 Morocco 0-1;
W vs #8 Switzerland 6-1;
L vs #4 France 0 (3)-(5) 0;

6. England: If you watch England play, the eye test will tell you that this team is not even a top 20 team on the planet. On the back of Southgate, England played defensive football that relied on miracles to win games. England got relegated in the Nations League, saw a round of 16 exit in the World Cup, and drew in regular time to Denmark, Slovenia, Slovakia, Switzerland, and had 90th minute winner over the Netherlands on their path to the Euro finals. Results keep them barely afloat, and a ranking of 6th almost feels generous.

7. Morocco: The Moroccans made it to the World Cup semifinals drawing Croatia and beating Belgium, Canada, Spain, and Portugal until they finally lost to France. Having one of the hardest paths, they proved they were among the world's best. However, a disappointing round of 16 loss in the CAF Cup of Nations drops them to 7th. FIFA punishes them more dropping them to 14th. A recent Olympic bronze medal tells that Morocco's story is not finished and that they are still among the world's best.

8. Switzerland: The Swiss are the constant characterization of a medium-level team. Round of 16 exit in the World Cup to Portugal, quarter final exit to England in the Euro, and group stage finish 2 points behind Spain in the Nations League. They deserve a high ranking due to their ability to compete, but that's about it for them. They compete to their eventual loss to one of the teams ranked above them. Their lack of ability to beat one of the teams above is why FIFA has them ranked 15. However, they have Italy ranked 10th when Switzerland just beat them 2-0 in the Euro R16.

9. Croatia: The Croats made the World Cup semifinals and have had back to back appearances there. They also were runner ups in the Nations League and are clearly among the world's best at their peak. However, they are appearing to slip a bit as their roster gets older and saw a disappointing group stage exit in the Euro. Their recent success still tells that they should not slip too far, but FIFA is already ahead of them and dropped them to 12.

10. Brazil: Brazil have disappointed in what feels like almost every single tournament since their last World Cup win. This team was favored to win the 2022 World Cup and saw a loss to Croatia in penalties in the quarter finals, they then drew to Costa Rica and lost to Uruguay to be eliminate in the quarter finals of the Copa American. They rank 6th in the current World Cup Qualifying table. Brazil is not what they once were and I am not sure what is keeping them afloat in the FIFA table.

L vs #9 Croatia 1 (2)-(4) 1;
T vs Costa Rica 0-0;
L vs #12 Uruguay 0 (2)-(4) 0;
W vs #20 South Korea 4-1;

17. Belgium: A tale of disappointment is Belgium. Always ranked among the best in FIFA yet never performing among the best. Anytime a ranking comes out, the most popular comment is usually, "how is Belgium still among the top?" Given my ranking, I am not sure as well. It likely is that Belgium beats up on inferior teams when they have the chance, but when it comes to tournaments, Beligum ghosts. They exited the group stage of the World Cup and nearly did so again at the Euro just to get eliminated in R16, and they got eliminated falling 6 points short of the Netherlands in the Nations League. Very undeserving of the 6th place ranking FIFA gives them.

Personal Nationalities:
20. USA: The United States would like to believe they are a competitive nation in the world now as they have what feels like a golden generation, but the results are just not coming. Sure, they are beating one of the worst Mexican teams ever over and over again, but it means nothing. They won their nations league, rested for the Gold Cup, and then got embarassed in the Copa America by Panama. Big changes are needed to not waste this World Cup cycle going into 2026.

21. Mexico: It is embarassment after embarassment for the Mexican team. In 2014, it felt as if this nation could shake the world being 2 minutes away from a World Cup quarterfinal appearance against rivals Costa Rica, and since then Mexico has never looked the same. In the recent cycle, Mexico couldn't exit the group stage when the group was relatively an easy group with the exception of Argentina. They repeated this embarassment by not exiting a group in the Copa America that they should have had no problem exiting. Mexico will now go into a world cup that they are hosting with extremely low expectations. If the Mexican federation doesn't begin prioritizing results and continues to prioritize short-term profits, this team will soon be left behind in the developing football world.

# Conclusion

While this project was made with the purpose of entertainment for myself, I found it to be a way of learning about data cleaning and manipulation to create a formula that can be followed to have a simpler and more direct approach to global FIFA Rankings. By implementing this method of ranking, not only will outdated results have less of an impact on current ranking, but it encourages teams to play at their most competitive level at each competition. Games from over four years ago should have minimal impact on the ranking status of each team.

This ranking system also gives equal opportunity for nations that perform at a high level in non Europe and South American confederations to compete in their tournaments and make a name for themselves. The current FIFA rankings allow for there to be too much influence from underperforming European nations such as Sweden ranking 29th in the FIFA World Rankings despite having not even qualified to either the last world cup or the Euro. They should therefore not rank over the current Asian champions, Qatar.